app/src/main/java/com/a0xffffffff/dinesum/RequestFeedFragment.java

```diff
@@ -0,0 +1,190 @@
+
+        mListView = (ListView) view.findViewById(R.id.requestfeed_request_list);
+        mProgressBar = (ProgressBar) view.findViewById(R.id.requestfeed_progressBar);
+
+        if (mDataReady) {
+            initListView();
+        } else {
+            mProgressBar.setVisibility(View.VISIBLE);
+            mListView.setVisibility(View.GONE);
+        }
+
+        return view;
+    }
+
+    public void updateListView() {
+        if (mRequests == null) {
+            mRequests = new ArrayList<>();
+        }
+        if (mRequests.equals(RequestTracker.getInstance().getNearbyRequests())) {
+            return;
+        }
+        mRequests.clear();
+        mRequests.addAll(RequestTracker.getInstance().getNearbyRequests());
+        mAdapter.notifyDataSetChanged();
+    }
+
+    public void initListView() {
+        mDataReady = true;
+        if (mProgressBar == null) {
+            return;
+        }
+
+        mProgressBar.setVisibility(View.GONE);
+
+        mRequests = RequestTracker.getInstance().getNearbyRequests();
+        mAdapter = new RequestAdapter(getActivity(), mRequests);
+
+        mListView.setAdapter(mAdapter);
+        mListView.setVisibility(View.VISIBLE);
+
+//        mListView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+//            @Override
+//            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+//                final Request request = (Request) adapterView.getItemAtPosition(i);
+//                final RequestData requestData = request.getRequestData();
+//                clickedRequest = request;
+//                final String requesterId = request.getRequesterID();
+//                final String requestStatus = request.getRequestState();
+//                final String restaurantId = requestData.getRestaurant().getRestaurantID();
+//                final String restaurantName = requestData.getRestaurant().getRestaurantName();
+//                final String restaurantAddress = requestData.getRestaurant().getRestaurantAddress();
+//                final String lunchTopic1 = requestData.getTopic1();
+//                final String lunchTopic2 = requestData.getTopic2();
+//
+//                final Intent intent = new Intent(getActivity(), RequestInfoActivity.class);
+//                intent.putExtra("requesterId", requesterId);
+//                intent.putExtra("requestStatus", requestStatus);
+//                intent.putExtra("restaurantId", restaurantId);
+//                intent.putExtra("restaurantName", restaurantName);
+//                intent.putExtra("restaurantAddress", restaurantAddress);
+//                intent.putExtra("lunchTopic1", lunchTopic1);
+//                intent.putExtra("lunchTopic2", lunchTopic2);
+//                startActivityForResult(intent, 0);
+//            }
+//        });
+    }
+
+//    @Override
+//    public void onActivityResult(int requestCode, int resultCode, Intent data) {

```

Comment:
Are we also going to have a state machine like this?

Reply:
So this is the stuff I was talking about yesterday. I'll have a utility class to handle the onActivityResult, and the implementation here simply just calls that utility class. Since the request info activity isn't implemented yet, I haven't done so. It'll be in the next PR